ori/volume/server/volume.go

```diff
@@ -16,235 +16,107 @@ package server
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"strings"
+	"time"
 
-	"github.com/google/uuid"
-	"github.com/onmetal/cephlet/ori/volume/apiutils"
-	"github.com/onmetal/cephlet/pkg/ceph"
-	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+	"github.com/go-logr/logr"
+	orimeta "github.com/onmetal/onmetal-api/ori/apis/meta/v1alpha1"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	userIDKey  = "userID"
-	userKeyKey = "userKey"
-
-	pvPoolKey               = "pool"
-	monitorConfigMapDataKey = "csi-cluster-config-json"
-	pvImageNameKey          = "imageName"
-	pvRadosNamespaceKey     = "radosNamespace"
+	MonitorsKey = "monitors"
+	ImageKey    = "image"
+	UserIDKey   = "userID"
+	UserKeyKey  = "userKey"
 )
 
-type AggregateCephVolume struct {
-	Pvc          *corev1.PersistentVolumeClaim
-	ImagePvc     *corev1.PersistentVolumeClaim
-	VolumeClass  *storagev1alpha1.VolumeClass
-	AccessSecret *corev1.Secret
+type PopulationImage struct {
+	Name string
+	Size uint64
 }
 
-var onmetalVolumeStateToORIState = map[corev1.PersistentVolumeClaimPhase]ori.VolumeState{
-	corev1.ClaimPending: ori.VolumeState_VOLUME_PENDING,
-	corev1.ClaimBound:   ori.VolumeState_VOLUME_AVAILABLE,
-	corev1.ClaimLost:    ori.VolumeState_VOLUME_ERROR,
+type CephImage struct {
+	Id          string
+	Annotations map[string]string
+	Labels      map[string]string
+	Generation  int64
+	CreatedAt   time.Time
+	DeletedAt   time.Time
+
+	Wwn                string
+	Pool               string
+	Size               uint64
+	PopulatedImageName string
+	Class              string
+	State              ori.VolumeState
```

Comment:
`State` should be specific to your implementation and not use `ori` types. Your internal types have to function without `ori`.